sql/rpl_gtid.cc

```diff
@@ -3607,6 +3634,31 @@ my_bool Id_delegating_gtid_event_filter<T>::has_finished()
          m_num_completed_filters == m_num_stateful_filters;
 }
 
+/**
+  Iteration block for Id_delegating_gtid_event_filter::verify_completed_state()
+*/
+static my_bool
+verify_subfilter_completed_state(void *entry, void *is_any_incomplete)
+{
+  if (entry && reinterpret_cast<
+    gtid_filter_element<decltype(rpl_gtid::domain_id)> *
+  >(entry)->filter->verify_completed_state())
+    *(my_bool *)is_any_incomplete= TRUE;
+  return FALSE; // do not terminate early
+}
+
+template <typename T>
+my_bool Id_delegating_gtid_event_filter<T>::verify_completed_state()
+{
+  if (has_finished()) // fast happy path
+    return FALSE;
+  // If a user-defined filters is not deactivated, it may not be complete.
+  my_bool is_any_incomplete= FALSE;
+  my_hash_iterate(&m_filters_by_id_hash,
+                  verify_subfilter_completed_state, &is_any_incomplete);
```

Comment:
Instead of iterating the hash ~~(and segfault myself from the hash’s untypedness)~~, I could’ve directly iterated the `Domain_gtid_event_filter::m_stop_filters`.
Iterating `Id_delegating_gtid_event_filter::m_filters_by_id_hash`
* future-proofs for `--start-position` warnings – if we want to.
* is possible in the superclass, so `Server_gtid_event_filter` (used by `--ignore-server-ids`) can inherit the method override.

Any preference?